### PR TITLE
Form Trigger Block and Shortcode

### PR DESCRIPTION
**Pull request by n7studios**
## Summary

Registers a Gutenberg block and Classic Editor shortcode, allowing the user to insert a button into a Page, Post or Custom Post, which triggers display of a ConvertKit modal, sticky or slide in form.

Gutenberg block:
https://www.loom.com/share/317afdc789c54a94b652cdc641fefb6f

Classic Editor shortcode:
https://www.loom.com/share/b9af851403f646b59cdbc934db8011fb

This saves the user having to manually copy/paste an e.g. modal form's trigger link (ConvertKit > Edit Form > Settings) and also copy/paste the necessary script (ConvertKit > Edit Form > Publish)
Screenshot 2023-04-14 at 11 56 47
Screenshot 2023-04-14 at 11 56 50

## Testing

- `PageBlockFormTriggerCest`: Test that the ConvertKit Form Trigger Block in Gutenberg works as expected
- `PageShortcodeFormTriggerCest`: Test that the [convertkit_formtrigger] shortcode works as expected

## Checklist

* [x] I have [written a test](TESTING.md#writing-an-acceptance-test) and included it in this PR
* [x] I have [run all tests](TESTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](TESTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](DEVELOPMENT.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](DEVELOPMENT.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

### CODE REVIEW

File: includes/blocks/class-convertkit-block-product.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public function enqueue_scripts() {
 	 */
 	public function enqueue_styles() {
 
-		wp_enqueue_style( 'convertkit-product', CONVERTKIT_PLUGIN_URL . 'resources/frontend/css/product.css', array(), CONVERTKIT_PLUGIN_VERSION );
```
Comment (noelherrick) [inline includes/blocks/class-convertkit-block-product.php:78]: What this intentional?

File: resources/backend/js/gutenberg-block-form-trigger.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * Custom callback function to render the ConvertKit Form Trigger Block preview in the Gutenberg Editor.
+ *
+ * @since 	2.2.0
+ */
+function convertKitGutenbergFormTriggerBlockRenderPreview( block, props ) {
+
+	// If no Form has been selected for display, return a prompt to tell the editor
+	// what to do.
+	if ( props.attributes.form === '' ) {
+		return wp.element.createElement(
```
Comment (noelherrick) [inline resources/backend/js/gutenberg-block-form-trigger.js:20]: I assume `wp` is a global on the WP frontend?

File: tests/_support/Helper/Acceptance/Plugin.php [hunk 3]
```diff
@@ -1000,7 +1066,7 @@ public function testRestrictContentDisplaysContent($I, $visibleContent = 'Visibl
 	public function seeProductOutput($I, $productURL, $text = false, $textColor = false, $backgroundColor = false)
 	{
 		// Confirm that the product stylesheet loaded.
-		$I->seeInSource('<link rel="stylesheet" id="convertkit-product-css" href="' . $_ENV['TEST_SITE_WP_URL'] . '/wp-content/plugins/convertkit/resources/frontend/css/product.css');
```
Comment (noelherrick) [inline tests/_support/Helper/Acceptance/Plugin.php:1003]: Just checking to see if this is intended.

**Review by noelherrick — APPROVED** [review]

File: includes/blocks/class-convertkit-block-product.php (hunk 1 above)
Comment (n7studios) [reply includes/blocks/class-convertkit-block-product.php:78]: Yes; this is renamed to `button.css` (line 79), as the same styles are now used for both the Form Trigger and Product blocks.

File: tests/_support/Helper/Acceptance/Plugin.php (hunk 3 above)
Comment (n7studios) [reply tests/_support/Helper/Acceptance/Plugin.php:1003]: Yes; this is renamed to `button.css`, as the same styles are now used for both the Form Trigger and Product blocks. The check on line 1004 is for this renamed stylesheet.

**Pull request merged**